src/org/exist/xquery/TryCatchExpression.java

```diff
@@ -343,22 +345,18 @@ private void addErrValue(final Throwable t) throws XPathException {
     // description is available (for example, if the error function 
     // was called with one argument).
     private void addErrDescription(final Throwable t, final ErrorCode errorCode) throws XPathException {
-        String description = errorCode.getDescription();
-        if (description == null && t != null) {
-            if(t instanceof XPathException) {
-                description = ((XPathException) t).getDetailMessage();
-            } else {
-                description = t.getMessage();
-            }
-        }
+        final Optional<String> errorDesc = Optional.ofNullable(errorCode.getDescription());
+        final Optional<String> throwableDesc = Optional.ofNullable(t instanceof XPathException ? ((XPathException) t).getDetailMessage() : t.getMessage());
+        final Sequence description = errorDesc
+                .<Sequence>map(
```

Comment:
i find these constructs very nice but makes for me unreadable ;-) for my code I typically avoid it....